core/src/main/java/org/apache/iceberg/ScanTaskParser.java

```diff
@@ -30,7 +30,10 @@ public class ScanTaskParser {
 
   private enum TaskType {
     FILE_SCAN_TASK("file-scan-task"),
-    DATA_TASK("data-task");
+    DATA_TASK("data-task"),
+    FILES_TABLE_TASK("files-table-task"),
+    ALL_MANIFESTS_TABLE_TASK("all-manifests-table-task"),
+    MANIFEST_ENTRIES_TABLE_TASK("manifest-entries-task");
```

Comment:
This is for the `BaseEntriesTable` task, which is the base class for `ManifestEntriesTable` and `AllEntriesTable`. Don't want to call it `base-entries-task` as it wouldn't be clear what kind of task.